xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/resources/BaseSearchResult.java

```diff
@@ -137,6 +155,63 @@ protected List<SearchResult> search(List<SearchScope> searchScopes, String keywo
         }
     }
 
+    private List<SearchResult> searchPagesSolr(List<SearchScope> searchScopes, String keywords, String wikiName,
+        String space, int number, int start, String orderField, String order, Boolean withPrettyNames,
+        Boolean isLocaleAware) throws QueryException, XWikiException
+    {
+        if (StringUtils.isBlank(keywords)) {
+            return List.of();
+        }
+
+        List<String> filterQueries = new ArrayList<>();
+        filterQueries.add("type:DOCUMENT");
+        filterQueries.add("hidden:false");
```

Comment:
I guess this should depend on the user preference. We don't have a generic hidden document query filter for Solr queries, do we?

Reply:
We don't. It would have to extend ``AbstractHiddenFilter``.